lib/datadog/search-syntax/src/vrl.rs

```diff
@@ -0,0 +1,592 @@
+        }
+    }
+}
+
+impl From<Comparison> for ast::Opcode {
+    fn from(c: Comparison) -> Self {
+        match c {
+            Comparison::Gt => ast::Opcode::Gt,
+            Comparison::Lt => ast::Opcode::Lt,
+            Comparison::Gte => ast::Opcode::Ge,
+            Comparison::Lte => ast::Opcode::Le,
+        }
+    }
+}
+
+impl From<ComparisonValue> for ast::Literal {
+    fn from(cv: ComparisonValue) -> Self {
+        match cv {
+            ComparisonValue::String(value) => value
+                .parse::<i64>()
+                .map(ast::Literal::Integer)
+                .unwrap_or_else(|_| ast::Literal::String(value)),
+
+            ComparisonValue::Numeric(value) => {
+                ast::Literal::Float(NotNan::new(value).expect("should be a float"))
+            }
+            ComparisonValue::Unbounded => panic!("unbounded values have no equivalent literal"),
+        }
+    }
+}
+
+/// Wrapper for a comparison value to be converted to a literal, with wrapped nodes.
+impl From<ComparisonValue> for ast::Node<ast::Expr> {
+    fn from(cv: ComparisonValue) -> Self {
+        make_node(ast::Expr::Literal(make_node(cv.into())))
+    }
+}
+
+/// Convert Datadog grammar to VRL.
+impl From<QueryNode> for ast::Expr {
+    fn from(q: QueryNode) -> Self {
+        match q {
+            // Match everything.
+            QueryNode::MatchAllDocs => nest_exprs(
+                make_queries(grammar::DEFAULT_FIELD)
+                    .into_iter()
+                    .map(|query| make_function_call("exists", vec![query])),
+                Opcode::Or,
+            ),
+            // Matching nothing.
+            QueryNode::MatchNoDocs => nest_exprs(
+                make_queries(grammar::DEFAULT_FIELD)
+                    .into_iter()
+                    .map(|query| make_not(make_function_call("exists", vec![query]))),
+                Opcode::Or,
+            ),
+            // Field existence.
+            QueryNode::AttributeExists { attr } => {
+                make_function_call("exists", vec![make_query(attr)])
+            }
+            QueryNode::AttributeMissing { attr } => {
+                make_not(make_function_call("exists", vec![make_query(attr)]))
+            }
+            // Equality.
+            QueryNode::AttributeTerm { attr, value }
+            | QueryNode::QuotedAttribute {
+                attr,
+                phrase: value,
+            } => make_function_call("match", vec![make_query(attr), make_regex(value)]),
+            // Comparison.
+            QueryNode::AttributeComparison {
+                attr,
+                comparator,
+                value,
+            } => make_op(make_node(make_query(attr)), comparator.into(), value.into()),
+            // Wildcard suffix.
+            QueryNode::AttributePrefix { attr, prefix } => make_function_call(
+                "match",
+                vec![make_query(attr), make_regex(format!("{}*", prefix))],
```

Comment:
maybe better use `starts_with`?

Reply:
I didn't use `starts_with` because of the word boundary issue we ran into.